[8.0] PXC-3609: Do not update binlog status vars when binlog is disabled

venkatesh-prasad-v opened the pull request:
https://jira.percona.com/browse/PXC-3609

Problem: In cluster mode, binlog related status vars are updated even
when binlog is disabled.

Solution: Do not register the status variables in the cache manager when
in binlog emulation mode (i.e, binlog is disabled)

venkatesh-prasad-v commented:
Jenkins: https://pxc.cd.percona.com/job/pxc-8.0-pipeline-parallel-mtr/95/console

Commit:
PXC-3609: Do not update binlog status vars when binlog is disabled
https://jira.percona.com/browse/PXC-3609

Problem: In cluster mode, binlog related status vars are updated even
when binlog is disabled.

Solution: Do not update status variables in the cache manager when
in binlog emulation mode (i.e, binlog is disabled)

kamil-holubicki reviewed (approved):
lgtm.
Please consider my concern about formatting.

kamil-holubicki commented on sql/binlog.cc:
```diff
@@ -12608,8 +12611,8 @@ static int show_binlog_vars(THD *thd, SHOW_VAR *var,
 
-static int show_binlog_snapshot_gtid_executed(
-    THD *thd, SHOW_VAR *var, char *buff [[maybe_unused]]) {
+static int show_binlog_snapshot_gtid_executed(THD *thd, SHOW_VAR *var,
```
If this changes formatting from upstream, I would leave it as it was.

Commit:
PXC-3609: Do not update binlog status vars when binlog is disabled

https://jira.percona.com/browse/PXC-3609

Merge remote-tracking branch 'origin/5.7-PXC-3609' into 8.0-PXC-3609

Problem: In cluster mode, binlog related status vars are updated even
when binlog is disabled.

Solution: Do not update status variables in the cache manager when
in binlog emulation mode (i.e, binlog is disabled)
